common/k8s/src/kube_stats/cluster_stats.rs

```diff
@@ -0,0 +1,70 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+pub struct ClusterStats {
+    pub resource: String,
+    pub r#type: String,
+    pub containers_init: i32,
+    pub containers_ready: i32,
+    pub containers_running: i32,
+    pub containers_terminated: i32,
+    pub containers_total: i32,
+    pub containers_waiting: i32,
+    pub cpu_allocatable: i64,
+    pub cpu_capacity: i64,
+    pub cpu_usage: i32,
+    pub memory_allocatable: i64,
+    pub memory_capacity: i64,
+    pub memory_usage: i64,
+    pub nodes_notready: i32,
+    pub nodes_ready: i32,
+    pub nodes_total: i32,
+    pub nodes_unschedulable: i64,
+    pub pods_allocatable: i64,
+    pub pods_capacity: i64,
+    pub pods_failed: i32,
+    pub pods_pending: i32,
+    pub pods_running: i32,
+    pub pods_succeeded: i32,
+    pub pods_unknown: i32,
+    pub pods_total: i32,
+}
+
+impl Default for ClusterStats {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ClusterStats {
+    pub fn new() -> ClusterStats {
+        ClusterStats {
+            containers_init: 0,
```

Comment:
wrapped in Option? to handle cases when metric is missing and excluded from serialization.

Reply:
all of the cluster stuff - or just usage/cap/allocatable?

I think zeros on the rest of the stats except for usage/allocatable/limits make sense